src/physics.F90

```diff
@@ -216,15 +231,73 @@ subroutine sample_photon_reaction(p)
 
           prob = prob + xs
           if (prob > cutoff) then
-            ! TODO: Create electron
-            ! E_electron = p % E - elm % shells(i_shell) % binding_energy
-
+            E_electron = p % E - elm % shells(i_shell) % binding_energy
+
+            ! Sample mu using non-relativistic Sauter distribution.
+            ! See Eqns 3.19 and 3.20 in "Implementing a photon physics
+            ! model in Serpent 2" by Toni Kaltiaisenaho
+            SAMPLE_MU1: do
+              r = prn()
+              if (FOUR * (ONE - r) * r >= prn()) then
+                rel_vel = sqrt(E_electron * (E_electron + TWO * MASS_ELECTRON))&
+                     / (E_electron + MASS_ELECTRON)
+                mu = (TWO * r + rel_vel - ONE) / &
+                     (TWO * rel_vel * r - rel_vel + ONE)
+                exit SAMPLE_MU1
+              end if
+            end do SAMPLE_MU1
+
+            phi = TWO*PI*prn()
+            uvw(1) = mu
+            uvw(2) = sqrt(ONE - mu*mu)*cos(phi)
+            uvw(3) = sqrt(ONE - mu*mu)*sin(phi)
+
+            ! Create secondary electron
+            call p % create_secondary(uvw, E_electron, ELECTRON, run_CE=.true.)
+
+            ! Allow electrons to fill orbital and produce auger electrons
+            ! and fluorescent photons
             call atomic_relaxation(p, elm, i_shell)
             p % event_MT = 533 + elm % shells(i_shell) % index_subshell
             p % alive = .false.
+            p % E = ZERO
+
             return
           end if
         end do
+
+        ! If no shell was sampled, give the whole phton energy to the electron.
+        ! See Eqn 3.9 in "Implementing a photon physics model in Serpent 2" by
+        ! Toni Kaltiaisenaho
+
+        ! Sample mu using non-relativistic Sauter distribution.
+        ! See Eqns 3.19 and 3.20 in "Implementing a photon physics
+        ! model in Serpent 2" by Toni Kaltiaisenaho
+        SAMPLE_MU2: do
+          r = prn()
+          if (FOUR * (ONE - r) * r >= prn()) then
+            rel_vel = sqrt(E_electron * (E_electron + TWO * MASS_ELECTRON))&
+                 / (E_electron + MASS_ELECTRON)
+            mu = (TWO * r + rel_vel - ONE) / &
+                 (TWO * rel_vel * r - rel_vel + ONE)
+            exit SAMPLE_MU2
+          end if
+        end do SAMPLE_MU2
+
+        phi = TWO*PI*prn()
+        uvw(1) = mu
+        uvw(2) = sqrt(ONE - mu*mu)*cos(phi)
+        uvw(3) = sqrt(ONE - mu*mu)*sin(phi)
+
+        ! Create secondary electron
+        call p % create_secondary(uvw, p % E, ELECTRON, run_CE=.true.)
+
+        ! TODO: figure out correct event_MT
+        p % event_MT = 533
```

Comment:
The masters thesis on photon transport in Serpent says they give the whole photon energy to the electron if no shell was sampled. Do you have any recommendation on what `event_MT` to assign to this interaction?